src/Features/CSharp/Portable/EditAndContinue/SyntaxComparer.cs

```diff
@@ -96,8 +96,6 @@ protected internal override bool TryGetParent(SyntaxNode node, [NotNullWhen(true
 
         protected internal override IEnumerable<SyntaxNode>? GetChildren(SyntaxNode node)
         {
-            Debug.Assert(HasLabel(node));
```

Comment:
>Debug.Assert(HasLabel(node)); [](start = 12, length = 29)

When is this not true? It'd odd because I'd think the tree diffing alg is only operating on labeled nodes and this should only be called from the alg.

Reply:
Same reason as https://github.com/dotnet/roslyn/pull/51021#discussion_r574216767, essentially there is no assumption/assertion that the root children are labelled for statement syntax.

Could add the `_compareStatementSyntax` to the assert if you'd prefer?

Reply:
Yes, I think we should apply the same assertion for statement level. LMK if anything fails doing so.

---
In reply to: [574252555](https://github.com/dotnet/roslyn/pull/51086#discussion_r574252555) [](ancestors = 574252555)

Reply:
Oops, sorry, missed this. Will file a follow up, there are lots of tests that fail with this assertion, but we can possibly move the check to CSharpEditAndContinueAnalyzer and have it only provide root children that are labelled.